src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -107,24 +109,147 @@ const columns = [
+  // TODO: Abstract pageParams object into a useable type.
+  updateEntries: (
+    group: boolean,
+    pageParams: { offset: number; pageSize: number },
+    filterParams: any
+  ) => void;
 };
 
-const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submissions }) => {
-  const dispatch = useDispatch();
+const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({
+  totalRows,
+  submissions,
+  updateEntries
+}) => {
+  // TODO: implement functionality for submission filtering by groups using the group state from useSession.
+
   const tableFilters = useTypedSelector(state => state.workspaces.grading.submissionsTableFilters);
 
   const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([
     ...tableFilters.columnFilters
   ]);
   const [globalFilter, setGlobalFilter] = useState<string | null>(tableFilters.globalFilter);
 
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  // This is needed because a filter change is accompanied with a page reset.
+  const resetPage = useCallback(() => setPage(0), [setPage]);
+
+  const [showAllSubmissions, setShowAllSubmissions] = useState(false);
+  const showSubmissionOptions = [
+    { value: false, label: 'ungraded' },
+    { value: true, label: 'all' }
+  ];
+
+  // TODO: implement isAdmin functionality
+  const [limitGroup, setLimitGroup] = useState(true);
+  const groupOptions = [
+    { value: true, label: 'my groups' },
+    { value: false, label: 'all groups' }
+  ];
+
+  // Dropdown tab options, which contains some external state.
+  // This can be a candidate for its own component once backend feature implementation is complete.
+  const pageSizeOptions = [
+    { value: 1, label: '1' },
+    { value: 5, label: '5' },
+    { value: 10, label: '10' },
+    { value: 20, label: '20' },
+    { value: 50, label: '50' }
+  ];
+
+  // modifies state setters to reset page as well.
+  function setStateAndReset<T>(stateChanger: React.Dispatch<T>): React.Dispatch<T> {
+    return (value: T) => {
+      stateChanger(value);
+      resetPage();
+    };
+  }
+
+  // Converts the columnFilters array into backend query parameters.
+  // Memoized as derived data to prevent infinite re-rendering.
+  // TEMP IMPLEMENTATION. Values currently hardcoded with knowledge of what a ColumnFilter is.
+  // TODO: remove hardcoding conversion of all submissions to column filter. it is a hacky workaround.
+  // TODO: implement reversible backend-frontend name conversion for use in RequestsSaga and here, remove hardcode.
+  // TODO: make a controller component, like in the achievements page, to handle conversion of page state into JSON.
+  const backendFilterParams = useMemo(() => {
+    return columnFilters
+      .concat(
+        showAllSubmissions
+          ? []
+          : [
+              {
+                id: 'gradingStatus',
+                value: GradingStatuses.none
+              }
+            ]
```

Comment:
Is this meant to replace

```ts
submissions.filter(
  s => showAllSubmissions || isSubmissionUngraded(s)
)
```

In the `Grading` component in the old version? If yes, note that `isSubmissionUngraded` is **not** check for grading status == none. Please refer to its implementation in `GradingUtils`.

Reply:
i see... isSubmissionsUngraded() checks for statuses none or grading + submitted. this may need subquerying, so i think i will add mention this for discussion this week.

Reply:
I don't think it needs subquerying. `GradingStatus` is a purely frontend construct. See

https://github.com/source-academy/frontend/blob/de886953eb3fe317aa8e915970fd95fb481abac7/src/commons/sagas/RequestsSaga.ts#L1362-L1377

To search for "ungraded only", you only need to check that:

* `isManuallyGraded === false`
* `submissionStatus === 'submitted'`
* `numGraded < numQuestions`

Btw, @GabrielCWT, the last bullet point is what the commented TODO was doing before the previous team shifted to client-side filtering.

Reply:
Yeah, an issue which we raised previously is that in order to find numGraded, we need to join answers with the submissions. But how can we guarentee that there is 10 submissions ungraded being sent over. What we have been doing so far is querying then limiting. If we apply the same logic to ungraded, we would join the entire answer table to submissions which is what caused long loading times.